#161966915 User Profiles and Update

leonardnjura opened the pull request:
#### What does this PR do?
- Allows users to create their profile

#### Description of Task to be completed?
> User logins
> User creates or updates their profile
> Time stamp is attached upon every profile edit

#### How should this be manually tested?
`$ git clone -b ft-user-profiles-161966915 https://github.com/andela/ah-infinity-stones` then
`$ cd ah-infinity-stones`
> Create and activate virtual env
virtualenv -p python3 venv
`$ source venv/bin/activate`
`$ pip3 install -r requirements.txt to install the dependencies`
> Setup environment variables. Copy all from below and paste them into the .env file. Update the values to your choice:
> Register user via `$ python3 manage.py createsuperuser --username john --email john.doe@example.com`

> `GET http://127.0.0.1:8000/api/profiles/`
> `GET http://127.0.0.1:8000/api/profiles/?page=2`
> `GET http://127.0.0.1:8000/api/profiles/u`
> `GET PUT http://127.0.0.1:8000/api/profiles/<username>`


```
export DATABASE_USER='user'
export DATABASE_PASSWORD='xxx'
export HOST='localhost'
export PORT='5432'
export DATABASE_TEST='ah_test'

 export EMAIL_HOST_USER='sendgrid_username'
 export EMAIL_HOST_PASSWORD='sendgrid_pswd'
 
```
> Activate the env variables by `$source .env`
> Run migrations: $ python3 manage.py migrate
> Run generic tests - $ python3 manage.py test
> Run specific tests python3 manage.py test authors/apps/profiles

#### What are the relevant pivotal tracker stories?
[#161966915](https://www.pivotaltracker.com/story/show/161966915)

#### Screenshots
<img width="1280" alt="screen shot 2018-12-14 at 10 40 14" src="https://user-images.githubusercontent.com/39657549/49989882-ac898c80-ff8c-11e8-967d-4ece10410488.png">
<img width="1280" alt="screen shot 2018-12-14 at 10 40 24" src="https://user-images.githubusercontent.com/39657549/49989883-ac898c80-ff8c-11e8-9800-fb650e7ac2d1.png">
<img width="1280" alt="screen shot 2018-12-14 at 10 40 31" src="https://user-images.githubusercontent.com/39657549/49989884-ad222300-ff8c-11e8-9816-c7cb5c6c6921.png">
<img width="1280" alt="screen shot 2018-12-14 at 10 40 40" src="https://user-images.githubusercontent.com/39657549/49989885-ad222300-ff8c-11e8-90a5-76433ff9ebac.png">

leonardnjura committed:
feat(userprofile): Users should be able to create their profiles

-  extends user profiles
-  authenticated users can edit profiles

[Delivers #161966915]

codeplus254 commented on authors/apps/profiles/urls.py:
```diff
@@ -0,0 +1,14 @@
+
+router = routers.DefaultRouter()
+
+urlpatterns = [
```
you ought to add the url path for confirmation email here as well :)

leonardnjura replied:
Ok i will refactor

rkemmy commented:
@leonardnjura the 'description of the task to be completed', 'what the PR does' and 'how to manually test this' are totally different and do not correlate to each other.

rkemmy commented:
@leonardnjura please check email_confirm.html line 19 and make the necessary adjustments

leonardnjura commented:
> @leonardnjura the 'description of the task to be completed', 'what the PR does' and 'how to manually test this' are totally different and do not correlate to each other.

r-wambui commented:
@leonardnjura please work on the merge conflicts also rebase develop to pick the current changes.

PromasterGuru commented:
Great work, work on the merge conflicts and rebase develope to ensure you have the latest changes.

r-wambui commented on authors/urls.py:
```diff
@@ -19,6 +19,9 @@
+        'rest_framework.urls', namespace='rest_framework')),
+    path('', include('authors.apps.articles.url')),
+    path('api/profiles/', include('authors.apps.profiles.urls')),
+    path('api/', include('authors.apps.authentication.urls')),
```
@leonardnjura Ensure we have the same naming of the urls. Liaise with the team on this.

r-wambui commented:
Resolve some of the tests, they are failing on travis but are passing develop travis build.

leonardnjura commented:
@r-wambui note I fixed the code climate issues and cleaned up codebase as requested. Build passes locally. I also added a followers field as needed by @codeplus254. Feel free to go through once more before merge

r-wambui commented on authors/apps/authentication/templates/email_confirm.html:
```diff
@@ -1,6 +1,7 @@
 
     <!doctype html>
     <html>
+<<<<<<< HEAD
```
@leonardnjura Why do we still have this?

leonardnjura replied:
Fixed it. Please check

leonardnjura commented:
@r-wambui kindly check the changes requested on this PR

ThaDeveloper commented:
To get or update a specific profile, is it `/profiles/u` or `profiles/<username>`? You could consider describing what the endpoints listed do.

r-wambui commented:
@leonardnjura Please ensure that travis can complete the build to show the tests passing on this branch

r-wambui commented on .travis.yml:
```diff
@@ -1,34 +1,47 @@
-  email: false
-
+# language: python
+# python:
```
@leonardnjura Why did you comment this?
